ts/src/pro/binance.ts

```diff
@@ -2143,6 +2153,181 @@ export default class binance extends binanceRest {
+        /**
+         * @method
+         * @name binance#watchPositions
+         * @description watch all open positions
+         * @param {[string]|undefined} symbols list of unified market symbols
+         * @param {object} params extra parameters specific to the binance api endpoint
+         * @returns {[object]} a list of [position structure]{@link https://docs.ccxt.com/en/latest/manual.html#position-structure}
+         */
+        await this.loadMarkets ();
+        await this.authenticate (params);
+        let market = undefined;
+        let messageHash = '';
+        if (!this.isEmpty (symbols)) {
+            market = this.getMarketFromSymbols (symbols);
+            messageHash = '::' + symbols.join (',');
+        }
+        let type = undefined;
+        [ type, params ] = this.handleMarketTypeAndParams ('watchPositions', market, params);
+        if (type === 'swap' || type === 'spot') {
+            type = 'future';
+        }
+        symbols = this.marketSymbols (symbols);
+        messageHash = type + ':positions' + messageHash;
+        const url = this.urls['api']['ws'][type] + '/' + this.options[type]['listenKey'];
+        const client = this.client (url);
+        this.setBalanceCache (client, type);
+        this.setPositionsCache (client, type, symbols);
+        const fetchPositionsSnapshot = this.handleOption ('watchPositions', 'fetchPositionsSnapshot', true);
+        const awaitPositionsSnapshot = this.safeValue ('watchPositions', 'awaitPositionsSnapshot', true);
+        const cache = this.positions[type];
+        if (fetchPositionsSnapshot && awaitPositionsSnapshot && this.isEmpty (cache)) {
+            return await client.future (type + ':fetchPositionsSnapshot');
+        }
+        const newPositions = await this.watch (url, messageHash, undefined, type);
+        if (this.newUpdates) {
+            return newPositions;
+        }
+        return this.filterBySymbolsSinceLimit (cache, symbols, since, limit, true);
+    }
+
+    setPositionsCache (client: Client, type, symbols: string[] = undefined) {
+        if (this.positions === undefined) {
+            this.positions = {};
+        }
+        if (type in this.positions) {
+            return;
+        }
+        this.positions[type] = new ArrayCacheBySymbolBySide ();
+        const fetchPositionsSnapshot = this.handleOption ('watchPositions', 'fetchPositionsSnapshot', false);
+        if (fetchPositionsSnapshot) {
+            const messageHash = type + ':fetchPositionsSnapshot';
+            if (!(messageHash in client.futures)) {
+                client.future (messageHash);
+                this.spawn (this.loadPositionsSnapshot, client, messageHash, type, symbols);
+            }
+        }
+    }
+
+    async loadPositionsSnapshot (client, messageHash, type, symbols) {
+        const positions = await this.fetchPositions (symbols, { 'type': type });
+        const cache = this.positions[type];
```

Comment:
@pcriadoperez binance (and some others exchanges) assume we have all the positions opened by default but in this case maybe we should filter them out? 
![image](https://github.com/ccxt/ccxt/assets/43336371/10e67410-61fd-4fb3-b17e-5bfebe2f564f)
Otherwise on every (non newUpdates) update we will return this huge array full of "inexistent" positions

Reply:
yes, I think it's cleaner, I'll filter them out